### PR TITLE
[core] Add stats for the gcs backend for telemetry.

### PR DESCRIPTION
**Pull request by fishbone**
Signed-off-by: Yi Cheng <chengyidna@gmail.com>

## Why are these changes needed?

To get better understanding of how GCS FT is used, adding this metrics.

Test:
```
cat /tmp/ray/session_latest/usage_stats.json
{"usage_stats": {"ray_version": "3.0.0.dev0", "python_version": "3.9.12", "schema_version": "0.1", "source": "OSS", "session_id": "70d3ecd3-5b16-40c3-9301-fd05404ea92a", "git_commit": "{{RAY_COMMIT_SHA}}", "os": "linux", "collect_timestamp_ms": 1660587366806, "session_start_timestamp_ms": 1660587351586, "cloud_provider": null, "min_workers": null, "max_workers": null, "head_node_instance_type": null, "worker_node_instance_types": null, "total_num_cpus": 16, "total_num_gpus": null, "total_memory_gb": 16.10752945020795, "total_object_store_memory_gb": 8.053764724172652, "library_usages": ["serve"], "total_success": 0, "total_failed": 13, "seq_number": 13, "extra_usage_tags": {"serve_api_version": "v1", "gcs_storage": "redis", "serve_num_deployments": "1"}, "total_num_nodes": 2, "total_num_running_jobs": 2}}
```

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/usage/usage_lib.py [hunk 1]
```diff
@@ -259,6 +259,9 @@ class TagKey(Enum):
     # The total number of running serve deployments as a string.
     SERVE_NUM_DEPLOYMENTS = auto()
 
+    # The GCS backed by a redis cluster.
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:262]: Could you mention the possible values here?

**Review by jjyao — COMMENTED** [review]
Could you add a simple test? Shouldn't be hard.

**Comment by jjyao** [comment]
@thomasdesr could you take a privacy review?

**Comment by iycheng** [comment]
> Could you add a simple test? Shouldn't be hard.

Do you have some sample code for this?

**Review by jjyao — APPROVED** [review]
Make sure to get @thomasdesr 's approval as well.

**Review by thomasdesr — COMMENTED** [review]
Is it possible to constrain the value's to our expected set? e.g. ("memory", "redis")

Its fine if not, just curious!

**Comment by iycheng** [comment]
@thomasdesr  I think it's possible (like we can do some validation). but I feel we probably need an update of the telemetry and make it a framework layer feature. The scope seems out of this PR. cc @jjyao

**Comment by jjyao** [comment]
Yea, when we register the key, we can also register all the valid values. But agree that it's out of scope of this PR.

**Comment by thomasdesr** [comment]
👍 Would be good candidate for the next time someone wants to work on the framework!

Being able to know if the values are bounded vs unbounded would be a good thing to have :D

**Pull request merged**